### PR TITLE
Basic Auth0 user functionality

### PR DESCRIPTION
**Pull request by kieran-pringle**
This is a work in progress that gets some basic Auth0 based user login sorted.

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -55,6 +58,10 @@ export default function Home() {
           <a href="https://www.buymeacoffee.com/karinakupp" target="_blank" rel="noreferrer">
             <div className={styles.link}>Support the project</div>
           </a>
+          <Link href="/api/auth/login">
+            <div className={styles.link}>Login</div>
```
Comment (kieran-pringle) [inline pages/index.tsx:62]: This is all that's needed for login! Obviously, some logic to go here with showing or hiding a logout link

File: pages/api/auth/[...auth0].tsx [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import { handleAuth } from '@auth0/nextjs-auth0';
+
+export default handleAuth();
```
Comment (kieran-pringle) [inline pages/api/auth/[...auth0].tsx:3]: This is what adds all the Auth0 callbacks. There was quite a bit of fiddling in the Auth0 account under the app for setting the expected URLs and such. Mayb e worth logging in there and looking.

Should we consider a separate dev/local and live Auth0 app? imo, yes. Would be a bit safer.

File: pages/index.tsx (hunk 1 above)
Comment (smoothpie) [reply pages/index.tsx:62]: nice! just checked, works great:) yeah, I'll handle the links display:)

File: pages/api/auth/[...auth0].tsx (hunk 2 above)
Comment (smoothpie) [reply pages/api/auth/[...auth0].tsx:3]: yeah, let's create separate apps. can you do this whenever you're free?:) 
I'll merge this as it is to dev now

Comment (kieran-pringle) [reply pages/api/auth/[...auth0].tsx:3]: Set a reminder for it for tomorrow. It was pretty simple.

**Pull request merged**